src/components/player.js

```diff
@@ -131,15 +133,26 @@ export default class Player extends PureComponent {
         logger.debug(ID.PLAYER, 'ready', ID.SCREENSHARE);
         this.player.screenshare = player;
         break;
+      case ID.EXTERNAL_VIDEOS:
+        logger.debug(ID.PLAYER, 'ready', ID.EXTERNAL_VIDEOS);
+        this.player.external_videos = player;
+        break;
       default:
         logger.debug('unhandled', media);
     }
 
-    if (this.player.video && this.player.screenshare) {
-      this.synchronizer = new Synchronizer(this.player.video, this.player.screenshare);
+    if (this.external_videos.length === 0) {
+      if (this.player.video && this.player.screenshare) {
+        this.synchronizer = new Synchronizer(this.player.video, this.player.screenshare);
+      }
+    } else {
+      if (this.player.video && this.player.screenshare && this.player.external_videos) {
+        this.synchronizer = new Synchronizer(this.player.video, this.player.screenshare, this.player.external_videos);
+      } 
     }
+    
   }
-
+  
```

Comment:
We can probably create a function to obtain the time stamp here.
`getCurrentPlayerTime() {`
`  const { time } = this.state;`
`  return time;`
`}`

And this function should be registered at the constructor like:
`this.getCurrentPlayerTime = this.getCurrentPlayerTime.bind(this);`